Mathlib/RingTheory/HahnSeries.lean

```diff
@@ -609,22 +609,38 @@ theorem order_one [MulZeroOneClass R] : order (1 : HahnSeries Γ R) = 0 := by
   · exact order_single one_ne_zero
 #align hahn_series.order_one HahnSeries.order_one
 
-instance [NonUnitalNonAssocSemiring R] : Mul (HahnSeries Γ R) where
-  mul x y :=
-    { coeff := fun a =>
-        ∑ ij in addAntidiagonal x.isPWO_support y.isPWO_support a, x.coeff ij.fst * y.coeff ij.snd
-      isPWO_support' :=
+variable {V : Type*} [AddCommMonoid V]
+
+instance instSMulHahn [Zero R] [SMul R V] : SMul (HahnSeries Γ R)
+    (HahnSeries Γ V) where
```

Comment:
This instance is a risky one, because if `V` is itself `HahnSeries` then there are two different actions.
This is the same as the reason why having `R[X]` act on `A[X]` is a bad idea, because then `R[X]` has two different actions on `R[X][X]` written the same way. I think it would probably be best not to have this instance, to match what we do with polynomials.

Reply:
I see.  Should I make a type alias, like in `Mathlib.Data.Polynomial.Module`?